### ISSUE TITLE
In TV_3D, what does weight_r(g or b) do?

### ISSUE THREAD

**Issue by tnt-ooo-tnt** [HuiZeng/Image-Adaptive-3DLUT#44]
In case of weight_r (deleted g or b statement for ease):

        self.weight_r = torch.ones(3,dim,dim,dim-1, dtype=torch.float)
        self.weight_r[:,:,:,(0,dim-2)] *= 2.0

    def forward(self, LUT):
        dif_r = LUT.LUT[:,:,:,:-1] - LUT.LUT[:,:,:,1:]
        tv = torch.mean(torch.mul((dif_r ** 2),self.weight_r)) + torch.mean(torch.mul((dif_g ** 2),self.weight_g)) + torch.mean(torch.mul((dif_b ** 2),self.weight_b))

Here, 1) I wonder why do we multiply dif_r ** 2 and self.weight_r instead of using just dif_r ** 2?
and 2) why do we multiply the edge elements of weight_r by 2.0?

I really appreciate to have this work :)

**Comment by HuiZeng** [comment]
Hi, the reason is the edge elements only calculate once in TV while the middle elements calculate twice (with before and after elements). I emperically found that the edge elements are not as smooth as the middle ones when visualizing the learned LUTs.
